### PR TITLE
add WebSocket RPC demo &&  add getCode, numberHash and totalTransactionCount interface for RPC

### PR DESCRIPTION
**Pull request by fqliao**

### CODE REVIEW

File: fisco-bcos/rpc/FakeModule.h [hunk 1]
```diff
@@ -205,8 +205,18 @@ class MockBlockChain : public BlockChainInterface
     dev::eth::LocalisedTransactionReceipt getLocalisedTxReceiptByHash(
         dev::h256 const& _txHash) override
     {
-        return LocalisedTransactionReceipt(
-            TransactionReceipt(), h256(0), h256(0), -1, Address(), Address(), -1, 0);
+        if (_txHash ==
+            jsToFixed<32>("0x7536cf1286b5ce6c110cd4fea5c891467884240c9af366d678eb4191e1c31c6f"))
```
Comment (wheatli) [inline fisco-bcos/rpc/FakeModule.h]: why the HASH ? Would you plz give some comments about it?

File: fisco-bcos/rpc/WebsocketServer.h [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        m_mapRpc.insert(std::make_pair("peers",
+            std::bind(&RpcFace::peersI, m_rpcFace, std::placeholders::_1, std::placeholders::_2)));
+        m_mapRpc.insert(
+            std::make_pair("groupPeers", std::bind(&RpcFace::groupPeersI, m_rpcFace,
+                                             std::placeholders::_1, std::placeholders::_2)));
+        m_mapRpc.insert(
+            std::make_pair("groupList", std::bind(&RpcFace::groupListI, m_rpcFace,
+                                            std::placeholders::_1, std::placeholders::_2)));
+        m_mapRpc.insert(
+            std::make_pair("getBlockByHash", std::bind(&RpcFace::getBlockByHashI, m_rpcFace,
```
Comment (wheatli) [inline fisco-bcos/rpc/WebsocketServer.h]: Should we give a standard of RPC API name?   Like as getXXX with a regular for all names
Make sure all the api methord are easily be remembered !

File: librpc/Rpc.cpp [hunk 3]
```diff
@@ -583,18 +616,18 @@ Json::Value Rpc::getTransactionReceipt(int _groupID, const std::string& _transac
                 JsonRpcException(RPCExceptionType::GroupID, RPCMsg[RPCExceptionType::GroupID]));
 
         h256 hash = jsToFixed<32>(_transactionHash);
-        auto tx = blockchain->getLocalisedTxByHash(hash);
-        auto txReceipt = blockchain->getTransactionReceiptByHash(hash);
-        if (tx.blockNumber() == 0 &&
-            tx.blockHash() ==
+        auto txReceipt = blockchain->getLocalisedTxReceiptByHash(hash);
```
Comment (wheatli) [inline librpc/Rpc.cpp]: I found that the blockNumber() return can be 0 or -1,  Plz tell me what do they means? and how to use them?
Give a method like isGenesisBlock(number) is a better way

File: fisco-bcos/rpc/FakeModule.h (hunk 1 above)
Comment (fqliao) [reply fisco-bcos/rpc/FakeModule.h]: Just for running rpc demo.

File: fisco-bcos/rpc/WebsocketServer.h (hunk 2 above)
Comment (fqliao) [reply fisco-bcos/rpc/WebsocketServer.h]: Yes, I have modified these api name.

File: librpc/Rpc.cpp (hunk 3 above)
Comment (fqliao) [reply librpc/Rpc.cpp]: I have used a macro definition.

**Pull request merged**